modules/ROOT/partials/configuration/ai_request.adoc

```diff
@@ -0,0 +1,176 @@
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${api_key}`
+        },
+        body: JSON.stringify(requestBody)
+      };
+      
+      // This function passes each new message into the plugin via the `streamMessage` callback.
+      const onMessage = (ev) => {
+        const data = ev.data;
+        if (data !== '[DONE]') {
+          const parsedData = JSON.parse(data);
+          const firstChoice = parsedData?.choices[0];
+          const message = firstChoice?.delta?.content;
+          if (message) {
+            streamMessage(message);
+          }
+        }
+      };
+      
+      const onError = (error) => {
+        // Stop operation and do not retry by the fetch-event-source
+        throw error;
+      };
+
+      // Use microsoft's fetch-event-source library to work around the 2000 character limit
+      // of the browser `EventSource` API, which requires query strings
+      return fetchApi
+      .then(fetchEventSource => 
+        fetchEventSource('https://api.openai.com/v1/chat/completions', {
+          ...openAiOptions,
+          openWhenHidden: true,
+          onmessage,
+          onerror
+        })
+      );
+    });
+  }
+});
+----
+
+[[request]]
+=== The `request` object
+
+The `+ai_request+` function is given a request object as the first parameter, which has these fields:
+
+`+query+`:: The submitted prompt as a string, without any context.
+
+`+context+`:: The current selection as a string, if any, or the current response displayed in the dialog. This can be combined with the `+query+`in a custom manner by the integrator to form a request. The current selection will be provided in HTML format, as will any displayed HTML response, and will increase token use.
+
+`+thread+`:: An array containing the history of requests and responses within the dialog, provided as an array of objects. This thread array is the same as is recorded in the xref:#getThreadLog[`+getThreadLog+` API], for current instance of the {pluginname} dialog.
+
+`+system+`:: An array of messages which provide instructions for handling the user prompts. The `+system+` array:
+
+[source, js]
+----
+[ 'Answer the question based on the context below.',
+  'The response should be in HTML format.',
+  'The response should preserve any HTML formatting, links, and styles in the context.' ]
+----
+
+`+prompt+`:: The submitted prompt as a string, combined with any current selection (when first opening the dialog) or the previous response. The {pluginname} plugin provides a customised format which combines these strings, though integrators are free to build their own with any of the other provided fields in the `+request+` object.
+
+[IMPORTANT]
+.The default prompt and token use.
+====
+The {pluginname} automatically prepends the `+system+` value as a string to the `+prompt+` value. The prompt also combines the `+query+` and `+context+` values into a single string.
```

Comment:
As someone who thought I understood OpenAI API, I find myself confused about context vs query. My guess in reading this, the context is the selected text, and the query is our prompt? 

But I don't link guessing...Perhaps a quick example might be helpful, just to get it across the finish line in people's heads.